nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/realm/ldap/LDAPRealm.java

```diff
@@ -189,52 +186,53 @@ public synchronized void init(Properties props) throws BadRealmException, NoSuch
 
         String url = props.getProperty(PARAM_DIRURL);
         String dn = props.getProperty(PARAM_USERDN);
-        String jaasCtx = props.getProperty(BaseRealm.JAAS_CONTEXT_PARAM);
+        String jaasCtx = props.getProperty(JAAS_CONTEXT_PARAM);
 
         if (isAnyNull(url, dn, jaasCtx)) {
-            throw new BadRealmException(sm.getString("ldaprealm.badconfig", url, dn, jaasCtx));
+            throw new BadRealmException(String.format(
+                "Incomplete configuration of ldap realm: url: {0} baseDN: {1} login module: {2}", url, dn, jaasCtx));
         }
 
-        this.setProperty(PARAM_DIRURL, url);
+        setProperty(PARAM_DIRURL, url);
         ldapBindProps.setProperty(Context.PROVIDER_URL, url);
-        this.setProperty(PARAM_USERDN, dn);
-        this.setProperty(BaseRealm.JAAS_CONTEXT_PARAM, jaasCtx);
+        setProperty(PARAM_USERDN, dn);
+        setProperty(JAAS_CONTEXT_PARAM, jaasCtx);
 
         String mode = props.getProperty(PARAM_MODE, MODE_DEFAULT);
         if (!MODE_DEFAULT.equals(mode)) {
-            throw new BadRealmException(sm.getString("ldaprealm.badmode", mode));
+            throw new BadRealmException(String.format("Unsupported mode {0}.", mode));
```

Comment:
here ... perhaps search for {0} in this diff.

Reply:
I made the same mistake everywhere 🫣